tests/test_nuclei_segmentation.py

```diff
@@ -118,3 +120,155 @@ def test_segment_nuclei_guassian_voting(self):
+                              scheduler='',
+                              stain_1='hematoxylin',
+                              stain_1_vector=[-1.0,
+                                              -1.0,
+                                              -1.0],
+                              stain_2='eosin',
+                              stain_2_vector=[-1.0,
+                                              -1.0,
+                                              -1.0],
+                              stain_3='null',
+                              stain_3_vector=[-1.0,
+                                              -1.0,
+                                              -1.0],
+                              style=None)
+
+    def test_image_inversion_flag_setter(self):
+        invert_image = nucl_det.image_inversion_flag_setter(self.args)
+        assert invert_image, True
+
+    def test_nuclei_detection(self):
+
+        # retrieve image from datastore
+        input_image_path = os.path.join(datastore.fetch('tcgaextract_ihergb.tiff'))
+        self.args.inputImageFile = input_image_path
+
+        # read the image
+        ts, is_wsi = nucl_det.read_input_image(self.args, process_whole_image=True)
+        it_kwargs = {
+            'tile_size': {'width': self.args.analysis_tile_size},
+            'scale': {'magnification': self.args.analysis_mag}
+        }
+
+        # determine number of nuclei
+        tile_fgnd_frac_list = nucl_det.process_wsi(ts, it_kwargs, self.args)
+        nuclei_list = nucl_det.detect_nuclei_with_dask(
+            ts,
+            tile_fgnd_frac_list,
+            it_kwargs,
+            self.args,
+            invert_image=False,
+            is_wsi=is_wsi,
+            src_mu_lab=None,
+            src_sigma_lab=None)
+        np.testing.assert_allclose(len(nuclei_list), 10, 1e+2)
+
+    def test_nuclei_detection_image_inverted(self):
+
+        # retrieve image from datastore
+        input_image_path = os.path.join(datastore.fetch('tcgaextract_ihergb.tiff'))
+        self.args.inputImageFile = input_image_path
+
+        # read the image
+        ts, is_wsi = nucl_det.read_input_image(self.args, process_whole_image=True)
+        it_kwargs = {
+            'tile_size': {'width': self.args.analysis_tile_size},
+            'scale': {'magnification': self.args.analysis_mag}
+        }
+
+        # determine number of nuclei
+        tile_fgnd_frac_list = nucl_det.process_wsi(ts, it_kwargs, self.args)
+        nuclei_list = nucl_det.detect_nuclei_with_dask(
+            ts,
+            tile_fgnd_frac_list,
+            it_kwargs,
+            self.args,
+            invert_image=True,
+            is_wsi=is_wsi,
+            src_mu_lab=None,
+            src_sigma_lab=None)
+        np.testing.assert_allclose(len(nuclei_list), 7497, 1e+2)
+
+    def test_nuclei_detection_frame_style(self):
+
+        # retrieve image from datastore
+        input_image_path = os.path.join(datastore.fetch('tcgaextract_ihergb.tiff'))
```

Comment:
This (also elsewhere), the `os.path.join()` outer function isn't doing anything -- `datastore.fetch` returns a single path and joining a single path is a no-operation.

Reply:
I agree, all these functions are updated now.